client/app/bundles/course/assessment/submission/components/FileInput.jsx

```diff
@@ -179,7 +234,10 @@ const FileInputField = (props) => {
 };
 
 FileInputField.propTypes = {
+  attachmentExists: PropTypes.bool,
+  attachmentType: PropTypes.oneOf(Object.values(AttachmentType)),
   name: PropTypes.string.isRequired,
+  isProgrammingQuestion: PropTypes.bool,
```

Comment:
Remember I have told you that this component is not supposed to know what the question type is? and also dont do too many if else. What happens if this component is used by another answer type in the future? 

You can create an adapter component in between the `Answer` component and `FileInput` (So `TextResponse` > `TextResponseFileInputField` > `FileInputField`. For example, create `TextResponseFileInput` that handles the `isMultipleUploadAllowed` and `disabled` logic then pass to this component.

Reply:
oh ya, sorry for that :(
upon further inspection, I found out that actually the attachmentType is only used to determine whether the file upload can still be done, and also whether the multiple upload is allowed. therefore, I have refactored this properly

Reply:
Why does FileInput need to know if there's any existing attachment? The job of the FileInputField is to input the file. It does not care if there's an existing attachment or not.

Reply:
simplified